### PR TITLE
Add analytic nature and account section fields

### PR DESCRIPTION
**Pull request by Badoche1337**
Hello,

This PR adds 2 specific fields : analytic nature (in order item) and account section (in order form). It provide two configuration options to make these 2 fields mandatory or not.

Have a nice day
Alan

### CODE REVIEW

File: inc/order_item.class.php [hunk 1]
```diff
@@ -383,6 +397,14 @@ public function showAddForm($plugin_order_orders_id) {
                ]);
                echo "</span></td>";
 
+               echo "<td>";
```
Comment (cedric-anne) [inline inc/order_item.class.php]: Should be displayed only if `isAnalyticNatureDisplayed`.

File: inc/order_item.class.php [hunk 2]
```diff
@@ -275,28 +287,29 @@ public function checkIFReferenceExistsInOrder($orders_id, $ref_id) {
    }
 
 
-   public function addDetails($ref_id, $itemtype, $orders_id, $quantity, $price, $discounted_price, $taxes_id) {
+   public function addDetails($ref_id, $itemtype, $orders_id, $quantity, $price, $discounted_price, $taxes_id, $analytic_nature_id) {
```
Comment (cedric-anne) [inline inc/order_item.class.php]: I guess calls on https://github.com/pluginsGLPI/order/blob/develop/front/order.form.php#L249 and https://github.com/pluginsGLPI/order/blob/develop/front/order.form.php#L268 have to be updated.

File: inc/accountsection.class.php [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+ ---------------------------------------------------------------------- */
+
+if (!defined('GLPI_ROOT')) {
+   die("Sorry. You can't access directly to this file");
+}
+
+class PluginOrderAccountSection extends CommonDropdown {
+   public static $rightname   = 'plugin_order_order';
+
+   public static function canCreate() {
```
Comment (cedric-anne) [inline inc/accountsection.class.php]: Are you sure thoose loosy permissions are necessary ?

File: inc/analyticnature.class.php [hunk 4]
```diff
@@ -0,0 +1,83 @@
...
+}
+
+class PluginOrderAnalyticNature extends CommonDropdown {
+   public static $rightname   = 'plugin_order_orders_item';
+
+   public static function getTypeName($nb = 0) {
+      return __("Analytic nature", "order");
+   }
+
+   public static function canCreate() {
```
Comment (cedric-anne) [inline inc/analyticnature.class.php]: Are you sure thoose loosy permissions are necessary ?

File: inc/order_item.class.php [hunk 5]
```diff
@@ -334,6 +347,7 @@ public function showAddForm($plugin_order_orders_id) {
                echo "<tr align='center'>";
                echo "<th>".__("Type")."</th>";
                echo "<th>".__("Product reference", "order")."</th>";
+               echo "<th>".__("Analytic nature", "order")."</th>";
```
Comment (cedric-anne) [inline inc/order_item.class.php]: Should be displayed only if `isAnalyticNatureDisplayed`.

File: inc/order_item.class.php [hunk 6]
```diff
@@ -544,6 +566,32 @@ public function showAddForm($plugin_order_orders_id) {
...
 
+   public function prepareInputForAdd($input) {
+      $config = PluginOrderConfig::getConfig();
+
+      if (isset($input["id"]) && $input["id"]>0) {
+         $input["_oldID"] = $input["id"];
+         unset($input['id']);
+         unset($input['withtemplate']);
+      } else {
+         if ($config->isAnalyticNatureMandatory() && $input["plugin_order_analyticnatures_id"] == 0) {
```
Comment (cedric-anne) [inline inc/order_item.class.php]: Should not be blocking unless `isAnalyticNatureDisplayed`.

File: inc/order_item.class.php [hunk 7]
```diff
@@ -544,6 +566,32 @@ public function showAddForm($plugin_order_orders_id) {
...
+            return [];
+         }
+      }
+
+      return $input;
+   }
+
+   public function prepareInputForUpdate($input) {
+      $config = PluginOrderConfig::getConfig();
+      if ($config->isAnalyticNatureMandatory() && $input["plugin_order_analyticnatures_id"] == 0) {
```
Comment (cedric-anne) [inline inc/order_item.class.php]: Should not be blocking unless `isAnalyticNatureDisplayed`.

File: inc/order_item.class.php [hunk 8]
```diff
@@ -768,6 +819,30 @@ public function getItems($rand, $data_ref, $plugin_order_orders_id, $numref, $ca
          } else {
             echo "<td align='center'>".$quantity."</td>";
          }
+
+         echo "<td align='center'>";
```
Comment (cedric-anne) [inline inc/order_item.class.php]: Should be displayed only if `isAnalyticNatureDisplayed`.

**Comment by cedric-anne** [comment]
I rebased, made some small fixes and remove modifications of the pot file (will de bone on release process). Be sure to update your branch before continue working on it.

**Comment by cedric-anne** [comment]
During my tests, I had some styling problems on the `Add to the order from the catalog` form. Looks like column sizes are made using `nth-child` targetting in CSS. It will not really work if column count is variable.

**Comment by Badoche1337** [comment]
Thank you for your fixes, I had WIP code that include same fixes as you but I haven't pushed it yet, sorry x)

**Pull request merged**